src/nvim/ex_docmd.c

```diff
@@ -9624,26 +9624,34 @@ static void ex_folddo(exarg_T *eap)
 
 static void ex_terminal(exarg_T *eap)
 {
-  char *name = (char *)p_sh;  // Default to 'shell' if {cmd} is not given.
-  bool mustfree = false;
-  char *lquote = "['";
-  char *rquote = "']";
+  char ex_cmd[1024];
 
+  // We will call termopen() with shell as an arglist if not given a {cmd}.
   if (*eap->arg != NUL) {
-    name = (char *)vim_strsave_escaped(eap->arg, (char_u *)"\"\\");
-    mustfree = true;
-    lquote = rquote = "\"";
+    char *name = (char *)vim_strsave_escaped(eap->arg, (char_u *)"\"\\");
+    snprintf(ex_cmd, sizeof(ex_cmd),
+             ":enew%s | call termopen(\"%s\") | startinsert",
+             eap->forceit ? "!" : "", name);
+    xfree(name);
+  } else {
+    char **argv = build_argv((char *)p_sh);
```

Comment:
Was there a specific reason why the new `build_argv(p_sh)` instead of `shell_build_argv(NULL,NULL)` was used?
```c
if (*p_sh == 0) {
   EMSG("No shell");
   return;
}

char **argv = shell_build_argv(NULL,NULL);
...
```